### DISCUSSION TITLE
Use ring-mqtt via python

### DISCUSSION THREAD

**Discussion by farbod1277** [tsightler/ring-mqtt#712] [Support Q&A] [answered]
Hi there,

I've currently got a python script running on my home server that I use to automate cameras and alarms. Unfortunately I couldn't find any python packages for ring that would allow me to arm/disarm the ring doorbell. I've now come across ring-mqtt but I'm not sure how I can use this without home-assistant. I'd like to use the arm/disarm functionality here within my python script. I was thinking of using paho-mqtt within python to communicate with this mqtt server but not sure how. Any guidance would be appreciated.

Thanks

**Comment by tsightler** [comment, marked as answer]
Interestingly, when I first started writing ring-mqtt, the entire goal was just to create some simple scripts for arming/disarming the alarm via CLI.  The ring-client-api project was by far the best library I could find, and it was a good opportunity to learn some Javascript and NodeJS, so I went that route.

Certainly you can use paho-mqtt to connect to the broker and send/receive commands.  The paho-mqtt library is very easy to use and overall setup would be simple, just install mosquitto MQTT broker, configure ring-mqtt to connect to it, and then using paho-mqtt from your python script to connect to the broker and send/receive commands.  There are plenty of examples for paho-mqtt around, so no need to duplicate that here.

As an alternative, for just sending arm/disarm commands, you can use the mosquitto client tools which can send commands via CLI.  You can see some examples of this in ring-mqtt in the scripts directory, the start-stream.sh script is a bash script that uses mosquitto_pub and mosquitto_sub to send/receive commands via MQTT to ring-mqtt (it's used by go2rtc to tell ring-mqtt to start/stop video streams).  This same approach would work with python.

Good luck!

**Reply by farbod1277** [reply]
Thanks you :)

**Reply by farbod1277** [reply]
Just as a follow up, I'm having trouble getting ring-mqtt to connect to the broker. I have allowed anonymous connections on the mosquitto conf file now but still get the ECONNREFUSED error.

**Reply by tsightler** [reply]
ECONNREFUSED is a pretty clear error, you will need to look at the logs on the broker and determine why it is refusing the connection.  Unfortunately, there's not much I can do to help you here, but again, there are plenty of resources out there just a Google search away.  MQTT is well discussed and documented and ring-mqtt doesn't do anything special, it uses a standard MQTT library used by 10's of thousands of projects and ring-mqtt itself has ~10,000 installes.  If you are having problems connecting to an MQTT broker you need to troubleshoot the broker settings and MQTT clients settings.  Good luck!
